### PR TITLE
Check license after login

### PR DESCRIPTION
**Pull request by efredriksson-modelon**
This PR uses the new version of the API where you can get the license of current user. If no license an exception is raised informing the user. I am very open to improvements to the error message; this was the best I could think of.

### CODE REVIEW

File: modelon/impact/client/client.py [hunk 1]
```diff
@@ -105,6 +105,13 @@ def __init__(
 
         self._sal.add_login_retry_with(api_key)
 
+        resp = self._sal.users.get_me()
+        if 'license' not in resp['data']:
+            raise exceptions.NoAssignedLicenseError(
+                "No assigned license during login. Either out of floating "
```
Comment (x-oflisback) [inline modelon/impact/client/client.py:111]: Maybe "sharable license" is the established term? At least that's what's used in the ops manual

**Review by x-oflisback — APPROVED** [review]
Nice, did not look too close at the test changes.

File: modelon/impact/client/client.py (hunk 1 above)
Comment (efredriksson-modelon) [reply modelon/impact/client/client.py:111]: We might decide to change the naming in the ops manual. But we should at least change this error message to use: "<floating|sharable> Deployment Add-on licenses" to be more precise.

Comment (efredriksson-modelon) [reply modelon/impact/client/client.py]: After discussion we went with floating. I also have made a commit where the type of the license is more precise.

**Review by andrewwren-modelon — APPROVED** [review]
:+1: lgtm

**Pull request merged**